rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -155,7 +155,11 @@ local function route_to_alternative_balancer(balancer)
 
   local header = ngx.var["http_" .. clean_target_header]
   if header then
-    if header == "always" then
+    if traffic_shaping_policy.headerValue then
+      if traffic_shaping_policy.headerValue == header then
+        return true
+      end
```

Comment:
The following two comments are related to not return `false` here when `traffic_shaping_policy.headerValue ~= header`.

IMO this is the right behaviour, but then we should change/add the tests to reflect that it fallback to cookie or weight if the header value is not equal to the configured header value.

Reply:
I have added a fallback test with cookie for a case when the sent header value doesn't match the annotated one.